sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/HiveTypeCoercion.scala

```diff
@@ -49,6 +49,7 @@ object HiveTypeCoercion {
       InConversion ::
       WidenSetOperationTypes ::
       PromoteStrings ::
+      SimplifyIntegerDecimalComparing ::
```

Comment:
We add this rule in analyser instead of optimizer because `DecimalPrecision` and `ImplicitTypeCasts` rules in analyser will add additional `Cast` operator to the integer expression.